### PR TITLE
MWPW-135380 Finishing Touches

### PR DESCRIPTION
**Pull request by Brandon32**
* Fix Banner exception
* Fix embed with text url
* Find all docx files
* Initial bulk update report

Resolves: [MWPW-135380](https://jira.corp.adobe.com/browse/MWPW-135380)

Test URLs:  [Top Blog Pages.xlsx](https://adobe-my.sharepoint.com/:x:/p/bmarshal/EQ-SUr3kukxMiJgO_yZkiPIBSnAC9oSnRiwkI0VyuVbfHg?e=8lFgxw)

### CODE REVIEW

File: bacom-blog/banner/banner.js [hunk 1]
```diff
@@ -8,11 +8,15 @@ export const FRAGMENTS_PATH = '/fragments';
...
+    return selectAllBlocks(mdast, 'Banner').map((block) => {
         const link = select('link', block);
-        const { pathname, search, hash } = new URL(link.url);
+        if (!link) {
+            return 'No link found in banner block';
+        }
+        const { pathname, search, hash } = new URL(link.url, 'https://main--bacom-blog--adobecom.hlx.page');
 
         if (pathname.includes(BANNERS_PATH)) {
             const fragmentUrl = `https://main--bacom-blog--adobecom.hlx.page${pathname.replace(BANNERS_PATH, FRAGMENTS_PATH)}${search}${hash}`;
```
Comment (meganthecoder) [inline bacom-blog/banner/banner.js:22]: Nit: Could pull `https://main--bacom-blog--adobecom.hlx.page` into a variable

File: bacom-blog/embed/embed.js [hunk 2]
```diff
@@ -18,13 +18,32 @@ const EMBED_URLS = [
...
+  }
+
+  const findUrl = (embedBlock) => {
+    const texts = selectAll('paragraph text', embedBlock);
+    const textUrl = texts.find(text => text?.value.includes('http'));
+    return textUrl ? createLink(textUrl) : null;
+  }
+
+  return selectAllBlocks(mdast, 'Embed').map((embedBlock) => {
+    const link = select('link', embedBlock) ? select('link', embedBlock) : findUrl(embedBlock);
```
Comment (meganthecoder) [inline bacom-blog/embed/embed.js:41]: Nit: Could pull `select('link', embedBlock)` into a variable

File: blog-migration.js [hunk 3]
```diff
@@ -43,30 +44,74 @@ async function updateSave(outputDocxFile, cached, mdast, sourceDocxFile, force,
...
     const reportDir = `${REPORT_DIR}/${PROJECT}`;
     const mdDir = `${MD_DIR}/${PROJECT}`;
     const docxDir = `${DOCX_DIR}/${PROJECT}`;
     const outputDir = `${output}/${PROJECT}`;
     const report = { succeed: [], skipped: [], failed: [], warned: [] };
-    // const entries = await loadIndex(PROJECT, `${SITE}${index}`, cached);
-    const top100 = await readIndex('bacom-blog/top-100.json');
-    const banners = await readIndex('bacom-blog/banners.json');
-    const entries = [...top100, ...banners];
+    const entries = (await readIndex(index));
```
Comment (meganthecoder) [inline blog-migration.js:108]: Parentheses?

File: blog-migration.js [hunk 4]
```diff
@@ -80,11 +125,13 @@ export async function main(index, cached, output, force) {
...
-            const migrationBlocks = tableMap.filter((block) => Object.keys(MIGRATION).includes(block.blockName));
+            const tableBlocks = tableMap.map((block) => block.blockName)
+            const migrationMap = tableMap.filter((block) => Object.keys(MIGRATION).includes(block.blockName));
+            const migrationBlocks = migrationMap.map((block) => block.blockName);
 
-            if (migrationBlocks.length === 0 && !entry.includes(BANNERS_PATH)) {
-                console.log('Skipping', entry, tableMap.map(block => block.blockName));
-                report.skipped.push(entry);
+            if (migrationMap.length === 0 && !entry.includes(BANNERS_PATH)) {
+                console.log(`${i}/${entries.length} Skipping ${entry}`, tableMap.map(block => block.blockName));
```
Comment (meganthecoder) [inline blog-migration.js:133]: Isn't this the same as your variable?

```suggestion
                console.log(`${i}/${entries.length} Skipping ${entry}`, tableBlocks);
```

**Review by meganthecoder — APPROVED** [review]
Looks good! Just a few nits

File: blog-migration.js (hunk 4 above)
Comment (Brandon32) [reply blog-migration.js:133]: Yes it is, thanks.

File: bacom-blog/embed/embed.js (hunk 2 above)
Comment (Brandon32) [reply bacom-blog/embed/embed.js:41]: Sounds good

File: blog-migration.js (hunk 3 above)
Comment (Brandon32) [reply blog-migration.js]: Not sure when that got added, removed.

**Review by meganthecoder — APPROVED** [review]

**Pull request merged**